### DISCUSSION TITLE
Tools and CI Meeting - Date 7/29

### DISCUSSION THREAD

**Discussion by makubacki** [tianocore/edk2#6001] [General]
# Status

Scheduled for 7/29 @ 4:30pm Pacific Time

# Agenda

1. Update: Reviewers automatically added to pull requests
   - [Completed] Basic support enabled in edk2
   - [In Progress] Workflow Enhancements (See PR for Details):
     - https://github.com/tianocore/edk2/pull/5992
3. Overview of quick changes to improve CI time
   - [Completed] Change to remove code coverage publish job at the end of the pipeline:
     - https://github.com/tianocore/edk2/pull/5978
   - [In Progress] Change to add code coverage pipeline publishing back:
     - https://github.com/tianocore/edk2/pull/6000
4. RFC: Bugzilla to GitHub issues
   - Details: https://github.com/tianocore/edk2/discussions/5926
   - @makubacki: Set a close data for comments on the RFC (give enough time for implementation)
   - @makubacki: Clarify in the template what "version" information is being asked for
   - @makubacki: Request Liming review the RFC in detail
6. CodeQL pin update
   - [In Progress] https://github.com/tianocore/edk2/pull/5986
7. Current status of enabling PRs in edk2-platforms
   - Deferred to next week to get an update from @bcran 

# Opens

- How does someone request a change to TianoCore Build documentation?
  - Instructions: https://github.com/tianocore-docs/edk2-TemplateSpecification/wiki/TianoCore-Documents-GitBook-Overview
  - Move remaining discussion on the particular need to an email on the use case
- Can this meeting be moved to a European friendly time?
  - Can potentially switch meetings between time zones on alternating weeks.
  - Likely range would 7am pacific to 8am pacific
  - Could propose a start around September

# Follow Up

- [ ] Next Meeting: Follow up on Bugzilla to GitHub Issue RFC
  - Mike Kinney's preference: Switch to GitHub by Monday, August 26, 2024 and switch the TianoCore Bugzilla to read only at that time
- [ ] Investigate making most jobs depend on a small Basic Acceptance Test (BAT) pool of checks to pass
- [ ] Can someone add measurements of CI information so we can make optimizations based on the data?
  - Example: Queue time, Queue depth, etc.
- [ ] Check the PR activity before pipeline is finished scenario. Does it cancel the current run?

# Todos From Previous Meeting

- Investigate GitHub merge queue as a replacement for Mergify.

# Groups.io meeting invite:

https://edk2.groups.io/g/devel/viewevent?repeatid=43269&eventid=2159616&calstart=2024-01-08

# Microsoft Teams meeting

Join on your computer or mobile app
[Click here to join the meeting](https://teams.microsoft.com/l/meetup-join/19%3ameeting_ZDI2ZDg4NmMtMjI1My00MzI5LWFmYjAtMGQyNjUzNTBjZGYw%40thread.v2/0?context=%7b%22Tid%22%3a%2272f988bf-86f1-41af-91ab-2d7cd011db47%22%2c%22Oid%22%3a%2223af6561-6e1c-450d-b917-d9d674eb3cb6%22%7d)
Or call in (audio only)
+1 425-616-0754,,330776518# United States, Seattle
Phone Conference ID: 330 776 518#
Find a local number | [Reset PIN](https://mysettings.lync.com/pstnconferencing)
[Learn More](https://aka.ms/JoinTeamsMeeting) | [Meeting options](https://teams.microsoft.com/meetingOptions/?organizerId=23af6561-6e1c-450d-b917-d9d674eb3cb6&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&threadId=19_meeting_ZDI2ZDg4NmMtMjI1My00MzI5LWFmYjAtMGQyNjUzNTBjZGYw@thread.v2&messageId=0&language=en-US)

**Comment by ghost** [comment]
> Current status of enabling PRs in edk2-platforms
>   Deferred to next week to get an update from @bcran

I've sent out a patch to create CODEOWNERS, REVIEWERS, CONTRIBUTORS and the GitHub workflow script to assign reviewers, and am currently waiting for people to review it.

**Comment by makubacki** [comment]
I created a discussion to track ideas for improving CI time - https://github.com/tianocore/edk2/discussions/6017.

Making an actual change will require a more detailed analysis of current performance. Feel free to comment or suggest anything in that discussion.
